src/macfc-security-hub-sync.ts

```diff
@@ -65,8 +72,83 @@ export class SecurityHubJiraSync {
     if (jiraConfig.jiraLabelsConfig) {
       this.jiraLabelsConfig = JSON.parse(jiraConfig.jiraLabelsConfig)
     }
+    if (securityHubConfig.consolidateTickets) {
+      this.jiraConsolidateTickets = securityHubConfig.consolidateTickets
+    }
+    if (jiraConfig.testFindings) {
+      this.testFindings = JSON.parse(jiraConfig.testFindings)
+      console.log('parsed', this.testFindings)
+    }
+  }
+  consolidateTickets(arr: SecurityHubFinding[]) {
+    const seen: GeneralObj = {} // Store unique titles
+    const finalList: SecurityHubFinding[] = []
+    arr.forEach(finding => {
+      const title = finding.title ?? ''
+      if (seen[title] >= 0) {
+        const i = seen[title]
+        finalList[i] = {
+          ...finalList[i],
+          Resources: [
+            ...(finalList[i].Resources ?? []),
+            ...(finding.Resources ?? [])
+          ]
+        }
+      } else {
+        const i = finalList.push(finding)
+        seen[title] = i - 1
+      }
+    })
+    return finalList
+  }
+  areSameLists(A: Resource[], B: Resource[]) {
+    if (A.length == B.length) {
+      let isSimilar = true
+      for (let i = 0; i < A.length; i = i + 1) {
+        let same = false
+        for (let j = 0; j < B.length && !same; j = j + 1) {
+          const a = A[i].Id ?? ''
+          const b = B[j].Id ?? ''
+          same = (a && b && a.includes(b)) as unknown as boolean
+        }
+        isSimilar = isSimilar && same
+      }
+      return isSimilar
+    }
+    return false
+  }
+  inAlreadyInNew(finding: SecurityHubFinding, List: SecurityHubFinding[]) {
```

Comment:
Should this be `isAlreadyInNew`?

Reply:
Good Catch - Renamed the function to "isAlreadyInNew"